diff -rupN include/armadillo_bits/config.hpp.cmake include/armadillo_bits/config.hpp.cmake
--- include/armadillo_bits/config.hpp.cmake	Fri Jan  3 01:59:32 2014
+++ include/armadillo_bits/config.hpp.cmake	Thu Feb 13 20:47:23 2014
@@ -60,7 +60,7 @@
 //// Make sure the directory has a trailing /
 
 #if !defined(ARMA_64BIT_WORD)
-// #define ARMA_64BIT_WORD
+#cmakedefine ARMA_64BIT_WORD
 //// Uncomment the above line if you require matrices/vectors capable of holding more than 4 billion elements.
 //// Your machine and compiler must have support for 64 bit integers (eg. via "long" or "long long")
 #endif
@@ -161,3 +161,16 @@
 #if defined(ARMA_DONT_USE_HDF5)
   #undef ARMA_USE_HDF5
 #endif
+
+#if defined(_MSC_VER)
+    #ifdef ARMA_DLL
+        #define ARMA_EXPORT __declspec(dllexport)
+    #elif defined(ARMA_STATIC_LIB)
+        #define ARMA_EXPORT
+    #else
+        #define ARMA_EXPORT __declspec(dllimport)
+    #endif
+#else
+    #define ARMA_EXPORT
+#endif
+
